h/search/index.py

```diff
@@ -33,24 +32,17 @@ def __init__(  # pylint: disable=too-many-arguments
         else:
             self._target_index = target_index
 
-    def index(self, annotation_ids=None, windowsize=PG_WINDOW_SIZE):
+    def index(self, annotation_ids: list, windowsize: int = PG_WINDOW_SIZE):
         """
         Reindex annotations.
 
         :param annotation_ids: a list of ids to reindex, reindexes all when `None`.
-        :type annotation_ids: collection
         :param windowsize: the number of annotations to index in between progress log statements
-        :type windowsize: integer
 
         :returns: a set of errored ids
         :rtype: set
         """
-        if annotation_ids is None:
-            annotations = _all_annotations(session=self.session, windowsize=windowsize)
```

Comment:
When no annos are passed, it gets them all from the DB.

This tries to be very efficient but I don't think is a feasible approach anymore.

Reply:
What even calls this `BatchIndexer` nowadays? Whatever it is should probably use the job queue and then we could get rid of this entirely